File: Solutions/1578/1578.go
package main

func minCost(colors string, neededTime []int) int {
	ans := 0
	maxNeededTime := neededTime[0]
	for i := 1; i < len(colors); i++ {
		if colors[i] == colors[i-1] {
			if maxNeededTime < neededTime[i] {
				ans += maxNeededTime
				maxNeededTime = neededTime[i]
			} else {
				ans += neededTime[i]
			}
		} else {
			maxNeededTime = neededTime[i]
		}
	}
	return ans
}

File: Solutions/1578/1578.js
function minCost(colors, neededTime) {
    let ans = 0;
    let maxNeededTime = neededTime[0];
    for (let i = 1; i < colors.length; ++i) {
        if (colors[i] === colors[i - 1]) {
            ans += Math.min(maxNeededTime, neededTime[i]);
            maxNeededTime = Math.max(maxNeededTime, neededTime[i]);
        } else {
            maxNeededTime = neededTime[i];
        }
    }
    return ans;
}
